src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/Profile.java

```diff
@@ -0,0 +1,387 @@
+  private Range<Double> floorAreaTerraceWeightRange;
+  private Range<Double> wifiSubjectiveValueWeight;
+  private Range<Double> pricePerNightSubjectiveValueWeight;
+  private Range<Double> nbMinNightSubjectiveValueWeight;
+  private Range<Double> teleSubjectiveValueWeight;
+
+  private Profile() {
+    this.linearAvf = null;
+    this.floorAreaWeightRange = null;
+    this.nbBedroomsWeightRange = null;
+    this.nbSleepingWeightRange = null;
+    this.nbBathroomsWeightRange = null;
+    this.terraceWeightRange = null;
+    this.floorAreaTerraceWeightRange = null;
+    this.wifiSubjectiveValueWeight = null;
+    this.pricePerNightSubjectiveValueWeight = null;
+    this.nbMinNightSubjectiveValueWeight = null;
+    this.teleSubjectiveValueWeight = null;
+  }
+
+  /**
+   * This function allows the user to clone an object Profile
+   *
+   * @return an object Profile
+   */
+  private Profile cloneProfile() {
+
+    Profile prof = new Profile();
+
+    prof.linearAvf = this.linearAvf;
+    prof.floorAreaWeightRange = this.floorAreaWeightRange;
+    prof.nbBedroomsWeightRange = this.nbBedroomsWeightRange;
+    prof.nbSleepingWeightRange = this.nbSleepingWeightRange;
+    prof.nbBathroomsWeightRange = this.nbBathroomsWeightRange;
+    prof.terraceWeightRange = this.terraceWeightRange;
+    prof.floorAreaTerraceWeightRange = this.floorAreaTerraceWeightRange;
+    prof.wifiSubjectiveValueWeight = this.wifiSubjectiveValueWeight;
+    prof.pricePerNightSubjectiveValueWeight = this.pricePerNightSubjectiveValueWeight;
+    prof.nbMinNightSubjectiveValueWeight = this.nbMinNightSubjectiveValueWeight;
+    prof.teleSubjectiveValueWeight = this.teleSubjectiveValueWeight;
+
+    return prof;
+  }
+
+  /**
+   * Gives the linearAVF of a Profile
+   *
+   * @return the LinearAVF of the Profile
+   */
+  public LinearAVF getLinearAVF() {
+    return this.linearAvf;
+  }
+
+  /**
+   * Gives the subjective value weight of a criterion
+   *
+   * @param crit the criterion we want to know the value
+   * @return the subjective value weight
+   */
+  public Range<Double> getWeightRange(Criterion crit) {
+    switch (crit) {
+      case TELE:
+        return teleSubjectiveValueWeight;
+      case TERRACE:
+        return terraceWeightRange;
+      case WIFI:
+        return wifiSubjectiveValueWeight;
+      case FLOOR_AREA:
+        return floorAreaWeightRange;
+      case FLOOR_AREA_TERRACE:
+        return floorAreaTerraceWeightRange;
+      case NB_BATHROOMS:
+        return nbBathroomsWeightRange;
+      case NB_BEDROOMS:
+        return nbBedroomsWeightRange;
+      case NB_SLEEPING:
+        return nbSleepingWeightRange;
+      case NB_MIN_NIGHT:
+        return nbMinNightSubjectiveValueWeight;
+      case PRICE_PER_NIGHT:
+        return pricePerNightSubjectiveValueWeight;
+      default:
+        throw new IllegalArgumentException();
+    }
+  }
+
+  /*public LinearAVF adaptWeightRange(Criterion crit, boolean upper) {
```

Comment:
Pourquoi cette partie du code est commentée ? Si elle n'est pas utile, il faudrait peut-être la supprimer.

Reply:
Fixed in commit 4012bf6